peachjam/resources.py

```diff
@@ -442,6 +443,20 @@ def get_case_numbers(row):
 
         return case_numbers
 
+    def before_import_row(self, row, **kwargs):
+        super().before_import_row(row, **kwargs)
+
+        if row.get("registry"):
```

Comment:
The registry should be the full code, such as `"kehc-high-court-at-foo"`. So your get_or_create() should use the code as the main lookup, and the other two fields as `defaults={}`.